### ISSUE TITLE
Deprecation in promise handling breaking code

### ISSUE THREAD

**Issue by joshuaSmith2021** [nodejs/help#1422]
* **Version**: v10.8.0
* **Platform**: Linux raspberrypi 4.14.52-v7+ #1123 SMP Wed Jun 27 17:35:49 BST 2018 armv7l GNU/Linux

I'm using firebase-tools, and I run `firebase login`, which throws this error:

```bash
(node:10422) UnhandledPromiseRejectionWarning: Error: Exited with code 3
    at ChildProcess.cp.once.code (/home/pi/.nvm/versions/node/v10.8.0/lib/node_modules/firebase-tools/node_modules/opn/index.js:84:13)
    at Object.onceWrapper (events.js:273:13)
    at ChildProcess.emit (events.js:182:13)
    at ChildProcess.EventEmitter.emit (domain.js:442:20)
    at maybeClose (internal/child_process.js:962:16)
    at Socket.stream.socket.on (internal/child_process.js:381:11)
    at Socket.emit (events.js:182:13)
    at Socket.EventEmitter.emit (domain.js:442:20)
    at Pipe._handle.close (net.js:599:12)
(node:10422) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:10422) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Comment by ankur-anand** [comment]
You should always handle promise rejection in your code. Since you've not handled the promise rejection you are getting this warning messssage.

Now Looking into the opn node_module line 84  I can see that, it uses child_process internally to spawn a new process,
```
const cp = childProcess.spawn(cmd, args, cpOpts);

	if (opts.wait) {
		return new Promise((resolve, reject) => {
			cp.once('error', reject);

			cp.once('close', code => {
				if (code > 0) {
					reject(new Error('Exited with code ' + code));
					return;
				}

				resolve(cp);
			});
		});
	}
```

So the error message `Error: Exited with code 3` is **coming from the exit code of the process, which according to official documentation means.**

> 3 Internal JavaScript Parse Error - The JavaScript source code internal in Node.js's bootstrapping process caused a parse error. This is extremely rare, and generally can only happen during development of Node.js itself.

I've never used firebase-cli  `firebase login` but looking into the **codebase of login in firebase-tool  i can see it tries to get open auth url with the `opn` module to get the Token and according to official doc `opn` Opens the url in the default browser**

```
var _loginWithoutLocalhost = function() {
....
open(authUrl);
....
}

var open = function(url) {
  opn(url).catch(function(err) {
    logger.debug("Unable to open URL: " + err.stack);
  });
};
```

So basically it may be error that has resulted from the platformon which you are trying to utilize the firebase cli

**Platform**: Linux raspberrypi 4.14.52-v7+ #1123 SMP Wed Jun 27 17:35:49 BST 2018 armv7l GNU/Linux 
 as the official Intro says **Kernel source tree for Raspberry Pi Foundation-provided kernel builds**

Check if Linux raspberrypi 4.14.52-v7 has any web browser.

**Comment by joshuaSmith2021** [comment]
It does have a web browser. Chromium is installed on it.

**Comment by joshuaSmith2021** [comment]
I've figured out my problem. It will be fixed in the next release of Firebase

**Comment by ankur-anand** [comment]
@joshuaSmith2021 Please share your solution, it might help someone who faces the similar issue.
